doc: fixing miscellaneous typos in RM

davidcok opened the pull request:
Fixes miscellaneous typos in RM, including some localized renumbering

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

cpitclaudel commented on docs/DafnyRef/Grammar.md:
```diff
@@ -565,16 +565,16 @@ datatype definitions.
 Here are some more examples.
 
-- `x: byte` (where `type byte = x | 0 <= x < 256`)
+- `x: byte` (where a value of type `byte` has values `x | 0 <= x < 256`)
```
I love everything except this change, because I'm not sure that "value `x | …`" is clear at this point in the manual.

davidcok replied:
Revised

cpitclaudel commented on docs/DafnyRef/Introduction.md:
```diff
@@ -26,12 +26,12 @@ checker—when the tool produces errors, the programmer responds by
 in the Dafny language.)
 
-The easiest way to try out Dafny is to [download](https://github.com/dafny-lang/dafny/releases) it
+The easiest way to try out `dafny` is to [download](https://github.com/dafny-lang/dafny/releases) it
```
?

cpitclaudel commented on docs/DafnyRef/Introduction.md:
```diff
@@ -26,12 +26,12 @@ checker—when the tool produces errors, the programmer responds by
+The easiest way to try out `dafny` is to [download](https://github.com/dafny-lang/dafny/releases) it
 to run it on your machine as you follow along with the [Dafny tutorial](../OnlineTutorial/guide).
-Dafny can be run from the command line (on Linux, MacOS, Windows or other platforms) or from an IDE
+The` dafny` tool can be run from the command line (on Linux, MacOS, Windows or other platforms) or from an IDE
```
```suggestion
The `dafny` tool can be run from the command line (on Linux, MacOS, Windows or other platforms) or from an IDE
```

davidcok replied:
Fixed

cpitclaudel commented on docs/DafnyRef/Modules.md:
```diff
@@ -104,7 +104,7 @@ of a single implicit unnamed global module.
 As described in the previous section, module declarations can be nested.
 It is also permitted to declare a nested module _outside_ of its
-"enclosing" module. So instead of
+"containing" module. So instead of
```
```suggestion
"parent" module. So instead of
```
Maybe?

davidcok replied:
The RM used 'parent' for inheritance; 'containing' works for syntactic containment

cpitclaudel commented on docs/DafnyRef/Modules.md:
````diff
@@ -840,77 +841,7 @@ use any local names:
-```dafny
-{% include_relative examples/Example-Refines1.dfy %}
-```
-The `A` in `refines A` refers to the submodule `A`, not the global `A`.
````
Was this part duplicated?

davidcok replied:
Yes

cpitclaudel commented on docs/DafnyRef/Types.md:
````diff
@@ -2915,14 +2916,6 @@ In contrast to one-dimensional arrays, there is no operation to
-  var intArr: array<int> := new int[] [1, 2, 3];
-  var intSeq: seq<int> := intArr[..];
-  assert intSeq == [1, 2, 3];
-```
````
Why is this part gone?

davidcok replied:
Duplicated in text at the end of 15.1, which is now the new 15.2

cpitclaudel commented on docs/DafnyRef/Introduction.md:
```diff
@@ -26,12 +26,12 @@ checker—when the tool produces errors, the programmer responds by
 in the Dafny language.)
 
-The easiest way to try out Dafny is to [download](https://github.com/dafny-lang/dafny/releases) it
+The easiest way to try out `dafny` is to [download](https://github.com/dafny-lang/dafny/releases) it
```
I understand this sentence as meaning that you try out the programming language, not the tool — hence the switch to `dafny` doesn't read too well

davidcok replied:
I'll revise a bit in a later PR

davidcok commented:
> On Jul 13, 2022, at 2:27 PM, Clément Pit-Claudel ***@***.***> wrote:
> 
> 
> @cpitclaudel requested changes on this pull request.
> 
> In docs/DafnyRef/Grammar.md <https://github.com/dafny-lang/dafny/pull/2401#discussion_r920378546>:
> 
> >  The common syntax that specifies the bound variables and what values they take on
>  is known as the *quantifier domain*; in the previous examples this is `x: nat | x <= 5`, 
>  which binds the variable `x` to the values `0`, `1`, `2`, `3`, `4`, and `5`.
>  
>  Here are some more examples.
>  
> -- `x: byte` (where `type byte = x | 0 <= x < 256`)
> +- `x: byte` (where a value of type `byte` has values `x | 0 <= x < 256`)
> I love everything except this change, because I'm not sure that "value x | …" is clear at this point in the manual.
> 
> 
I’ll elaborate
> 
> In docs/DafnyRef/Introduction.md <https://github.com/dafny-lang/dafny/pull/2401#discussion_r920378794>:
> 
> >  in the Dafny language.)
>  
> -The easiest way to try out Dafny is to [download](https://github.com/dafny-lang/dafny/releases) it
> +The easiest way to try out `dafny` is to [download](https://github.com/dafny-lang/dafny/releases) it
> ?
> 
> 
As stated elsewhere, the RM (mostly) uses Dafny for the language and `dafny` for the tool

> 
> In docs/DafnyRef/Introduction.md <https://github.com/dafny-lang/dafny/pull/2401#discussion_r920378973>:
> 
> >  to run it on your machine as you follow along with the [Dafny tutorial](../OnlineTutorial/guide).
> -Dafny can be run from the command line (on Linux, MacOS, Windows or other platforms) or from an IDE
> +The` dafny` tool can be run from the command line (on Linux, MacOS, Windows or other platforms) or from an IDE
> ⬇️ Suggested change
> -The` dafny` tool can be run from the command line (on Linux, MacOS, Windows or other platforms) or from an IDE
> +The `dafny` tool can be run from the command line (on Linux, MacOS, Windows or other platforms) or from an IDE

Right.

> 
> In docs/DafnyRef/Modules.md <https://github.com/dafny-lang/dafny/pull/2401#discussion_r920379286>:
> 
> > @@ -104,7 +104,7 @@ of a single implicit unnamed global module.
>  
>  As described in the previous section, module declarations can be nested.
>  It is also permitted to declare a nested module _outside_ of its
> -"enclosing" module. So instead of
> +"containing" module. So instead of
> ⬇️ Suggested change
> -"containing" module. So instead of
> +"parent" module. So instead of
> Maybe?
> 
> 
“Parent” is used for inheritance, whereas this is syntactic containment. So I’d rather “containing”. There was a mix of enclosing and containing, so I moved toward ‘containing'

> 
> In docs/DafnyRef/Modules.md <https://github.com/dafny-lang/dafny/pull/2401#discussion_r920381414>:
> 
> > -dependency.
> -
> -3. Step 2 is iterated for each ``NameSegment`` in the qualified module id,
> -   resulting in a module that is the final resolution of the complete
> -   qualified id.
> -
> -Ordinarily a module must be _imported_ in order for its constituent
> -declarations to be visible inside a given module `M`. However, for the
> -resolution of qualified names this is not the case.
> -
> -Ths example shows that the resolution of the refinement parent does not
> -use any local names:
> -```dafny
> -{% include_relative examples/Example-Refines1.dfy %}
> -```
> -The `A` in `refines A` refers to the submodule `A`, not the global `A`.
> Was this part duplicated?
> 
> 
Yes
> 
> In docs/DafnyRef/Types.md <https://github.com/dafny-lang/dafny/pull/2401#discussion_r920382202>:
> 
> > @@ -2915,14 +2916,6 @@ In contrast to one-dimensional arrays, there is no operation to
>  convert stretches of elements from a multi-dimensional array to a
>  sequence.
>  
> -## 15.3. Converting arrays to sequences {#sec-array-to-seq}
> -
> -To convert an array `x` to a sequence, use the `[..]` operator:
> -```dafny
> -  var intArr: array<int> := new int[] [1, 2, 3];
> -  var intSeq: seq<int> := intArr[..];
> -  assert intSeq == [1, 2, 3];
> -```
> Why is this part gone?
> 
> 

Because it was duplicative (though not verbatim) with the end of section 15.1, which was a better and longer explication; so I moved the heading up to be 15.2 and removed anything duplicative. 
> —
> Reply to this email directly, view it on GitHub <https://github.com/dafny-lang/dafny/pull/2401#pullrequestreview-1037789118>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABITDQE2WRY2YDMWC3P6E4TVT4DB7ANCNFSM53PTZVCA>.
> You are receiving this because you authored the thread.
>
